core/src/main/java/net/opentsdb/query/router/TimeRouterFactory.java

```diff
@@ -115,7 +115,7 @@ public void setupGraph(final QueryPipelineContext context,
     if (context.query().isDebugEnabled()) {
       StringBuilder buf = new StringBuilder();
       for (int i = 0; i < sources.size(); i++) {
-        if (i >= 0) {
+        if (i >= 0 && buf.length() > 0) {
```

Comment:
`i > 0` should be enough here

Reply:
should but I had a funky empty string popping up.